jwst/lib/set_telescope_pointing.py

```diff
@@ -148,6 +144,77 @@ def update_wcs(model, default_pa_v3=0., siaf_path=None, **kwargs):
         Keyword arguments used by matrix calculation routines.
     """
 
+    # If the type of exposure is not FGS, then attempt to get pointing
+    # from telemetry.
+    if model.meta.exposure.type.lower().startswith('fgs'):
+        update_wcs_from_header(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+    else:
+        update_wcs_from_telem(
+            model, default_pa_v3=default_pa_v3, siaf_path=siaf_path, **kwargs
+        )
+
+
+def update_wcs_from_header(model, default_pa_v3=0., siaf_path=None, **kwargs):
+    """ Update WCS pointing from header information
+
+    For FGS-like observations, nearly all information is already populated
+    except for the CD matrix. This simply calculates that matrix.
+
+    Parameters
+    ----------
+    model : `~jwst.datamodels.DataModel`
+        The model to update.
+    default_pa_v3 : float
+        If pointing information cannot be retrieved,
+        use this as the V3 position angle.
+    siaf_path : str
+        Unused and present only for API compatibility.
+
+    kwargs: dict
+        Unused and present only for API compatibility.
+    """
+
+    logger.info('Updating WCS from headers.')
+
+    # Get position angle
+    try:
+        pa = model.meta.wcsinfo.pa_v3
+    except AttributeError:
+        pa = default_pa_v3
+    pa_rad = pa * D2R
+
+    model.meta.wcsinfo.pc1_1 = -np.cos(pa_rad)
+    model.meta.wcsinfo.pc1_2 = np.sin(pa_rad)
+    model.meta.wcsinfo.pc2_1 = np.sin(pa_rad)
+    model.meta.wcsinfo.pc2_2 = np.cos(pa_rad)
```

Comment:
Is this the correct angle rotation? Usually the `-` sign goes with one of the `sin` functions depending on the direction.

Reply:
That's exactly the same as the way it's currently computed for science exposures in the `update_wcs` function. So maybe they're both wrong?

Reply:
This came from further past in the history, but will go over the docs to make sure.

Reply:
However, the question I have here is whether `vparity` should play a part here. For science, `vparity` is incorporated into the full transformation path. However, since we're short-circuiting here, I've got a sneaky suspicion that it needs to be included. I'll go through the transform logic, but was hoping the answer was already known 😄

Reply:
Also, about the science version of the transform: Have we not already had confirmation through archive/COAM with their preview code? I presume they are using the FITS WCS for that work?

Reply:
An example of using the parity is here

https://github.com/spacetelescope/stwcs/blob/master/stwcs/updatewcs/makewcs.py#L185

Reply:
As far as confirmation from CAOM is concerned, it's not obvious to me that we've ever yet had any consistent and/or reliable WCS info go all the way through the whole system to allow that kind of confirmation. Most of the test data end up with a rotation of zero and hence the PC matrix is just 1's and 0's, so there's been no confirmation of whether the rotation implied by the PC is correct. The most that's been verified is that when we set PC1_1 = -1, PC2_2 = 1 (and the cross terms to 0) images display properly with North up and East to the left.

Reply:
Discussed with Colin Cox and he says that VIdlParity should be included. He notes that for guiding, the OSS apertures will be used where VIdlParity=1. Regardless, added in VIdlParity inclusion.

Reply:
I'll add that VPARITY for all instruments, including FGS, is -1.  This means the sense of rotation of the V2/V3 frame is the opposite of that of the science X/Y frames, i.e. positive rotation of one means a negative rotation in the other.